src/commands/backup.rs

```diff
@@ -208,87 +217,116 @@ impl BackupCmd {
             }
+            if is_err {
+                Err(anyhow!("Not all snapshots were successfully generated!"))
+            } else {
+                Ok(())
+            }
+        })
+    }
 
-            // merge "backup" section from config file, if given
-            opts.merge(config.backup.clone());
-
-            let backup_opts = BackupOptions::default()
-                .stdin_filename(opts.stdin_filename)
-                .stdin_command(opts.stdin_command)
-                .as_path(opts.as_path)
-                .parent_opts(opts.parent_opts)
-                .ignore_save_opts(opts.ignore_save_opts)
-                .ignore_filter_opts(opts.ignore_filter_opts)
-                .no_scan(opts.no_scan)
-                .dry_run(config.global.dry_run);
-            let snap = repo.backup(&backup_opts, &sources, opts.snap_opts.to_snapshot()?)?;
-
-            if opts.json {
-                let mut stdout = std::io::stdout();
-                serde_json::to_writer_pretty(&mut stdout, &snap)?;
-            } else if opts.long {
-                let mut table = table();
-
-                let add_entry = |title: &str, value: String| {
-                    _ = table.add_row([bold_cell(title), Cell::new(value)]);
-                };
-                fill_table(&snap, add_entry);
-
-                println!("{table}");
-            } else if !opts.quiet {
-                let summary = snap.summary.unwrap();
-                println!(
-                    "Files:       {} new, {} changed, {} unchanged",
-                    summary.files_new, summary.files_changed, summary.files_unmodified
-                );
-                println!(
-                    "Dirs:        {} new, {} changed, {} unchanged",
-                    summary.dirs_new, summary.dirs_changed, summary.dirs_unmodified
-                );
-                debug!("Data Blobs:  {} new", summary.data_blobs);
-                debug!("Tree Blobs:  {} new", summary.tree_blobs);
-                println!(
-                    "Added to the repo: {} (raw: {})",
-                    bytes_size_to_string(summary.data_added_packed),
-                    bytes_size_to_string(summary.data_added)
-                );
-
-                println!(
-                    "processed {} files, {}",
-                    summary.total_files_processed,
-                    bytes_size_to_string(summary.total_bytes_processed)
-                );
-                println!("snapshot {} successfully saved.", snap.id);
+    fn backup_snapshot<P: ProgressBars, S: IndexedIds>(
+        mut self,
+        source: PathList,
+        repo: &Repository<P, S>,
+    ) -> Result<()> {
+        let config = RUSTIC_APP.config();
+        let snapshot_opts = &config.backup.snapshots;
+        if let Some(path) = &self.as_path {
+            // as_path only works in combination with a single target
+            if source.len() > 1 {
+                bail!("as-path only works with a single target!");
```

Comment:
```suggestion
                bail!("as-path only works with a single target profile!");
```

Reply:
My thinking is, that it is a user-facing error and it might make sense to stay consistent to talk about `target profiles` rather than `targets` (which is probably more understandable for us devs).

Reply:
Actually, this is not the target as used in `copy` here. But the target path set by `--as-path` for a source path. But the message is wrong anyway: It should be: `as-path only works with a single source`! I'll change it.